### PR TITLE
#4643: Home screen: replacing the search icon by a filter icon in the menu

### PR DESCRIPTION
**Pull request by mnaturel**
## Type of change

- [ ] Feature
- [ ] Bugfix
- [ ] Technical
- [x] Other : Enhancement

## Content

Replacing the search icon in the home menu by a filter icon to be more consistent.

## Motivation and context

See #4643 

## Screenshots / GIFs

4643-home-menu-light4643-home-menu-dark

## Tests

- Go to home screen
- Check a filter icon is displayed in the top right menu
- Click on the filter icon
- Check the filter screen is opened

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s): Android 11

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [x] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [x] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

**Review by ouchadam — APPROVED** [review]

File: changelog.d/4643.misc [hunk 1]
```diff
@@ -0,0 +1 @@
+Replacing search icon by filter icon
```
Comment (ouchadam) [inline changelog.d/4643.misc:1]: might be worth mentioning where the icon is replaced

Comment (mnaturel) [reply changelog.d/4643.misc]: Yes indeed! I have added more details.

**Comment by bmarty** [comment]
Looks like we are reverting what have been done for https://github.com/vector-im/element-android/issues/1384. The filter icon was not clear for the users.

Happy to accept the change, but I want to be sure that @daniellekirkwood has seen https://github.com/vector-im/element-android/issues/1384

**Comment by mnaturel** [comment]
> Looks like we are reverting what have been done for #1384. The filter icon was not clear for the users.
> 
> Happy to accept the change, but I want to be sure that @daniellekirkwood has seen #1384

Okay no problem. I have kept my work of moving the filter bar in the lists in local if needed.

**Comment by daniellekirkwood** [comment]
Have seen the comments and the previous discussion. Let's stay accurate (filter icon) and implement a search feature as soon as we can

**Pull request merged**